packages/vscode-extension/src/utilities/subprocess.ts

```diff
@@ -2,18 +2,59 @@ import { Logger } from "../Logger";
+import { getAppRootFolder } from "./extensionContext";
+const nodeExec = util.promisify(bareExec);
 
 export type ChildProcess = ExecaChildProcess<string>;
 
-function overridePWD<T extends execa.Options>(options?: T) {
+export async function getPathEnv() {
+  // We run an interactive shell session to make sure that tool managers (nvm,
+  // asdf, mise, etc.) are loaded and PATH is set correctly. Mise in
+  // particular sets the PATH by hooking into change dir and prompt display
+  // events.
+  // We make sure of that by doing a cd into app root directory, which has
+  // ensures that correct versions of tools are used (you can have different
+  // versions based on directory in most managers).
+
+  // Fish, bash, and zsh all support -i and -c flags.
+  const shellPath = process.env.SHELL ?? "/bin/zsh";
+  const appRoot = getAppRootFolder();
+  const { stdout: path } = await nodeExec(`${shellPath} -i -c 'cd "${appRoot}" && echo "$PATH"'`);
+  return path.trim();
+}
+
+let pathEnv: string | undefined;
+export async function setupPathEnv() {
+  if (!pathEnv) {
+    pathEnv = await getPathEnv();
+    if (!pathEnv) {
+      throw new Error("Error in getting PATH environment variable");
+    }
+  }
+}
+
+function overrideEnv<T extends execa.Options>(options?: T): T | undefined {
   // Some processes rely on PWD environment variable to tell the current working
-  // directory in which cases PWD takes precedence over process.cwd.
-  // By default, execa would copy all process env to the subprocess (which is desired)
-  // including PWD that may point to a different location that the selected cwd (indicated
-  // by options.cwd). Specifically, when VSCode is launched from the launcher (as opposed to
-  // being launched from command line using 'code' command), the PWD is set to "/".
-  // This method overrides PWD to the current cwd option when it's set for the subprocess call
-  // therefore removing the risk of the subprocess using the wrong working directory.
+  // directory in which cases PWD takes precedence over process.cwd. By default,
+  // execa would copy all process env to the subprocess (which is desired)
+  // including PWD that may point to a different location that the selected cwd
+  // (indicated by options.cwd). Specifically, when VSCode is launched from the
+  // launcher (as opposed to being launched from command line using 'code'
+  // command), the PWD is set to "/". This method overrides PWD to the current
+  // cwd option when it's set for the subprocess call therefore removing the
+  // risk of the subprocess using the wrong working directory.
+
+  // Additionally, we overwrite PATH env variable using env from interactive
+  // shell, ensuring that we have access to node and other tools, even when
+  // VSCode is launched as an application and not from the terminal.
+  if (pathEnv) {
+    options = {
+      ...options,
+      env: { ...options?.env, PATH: pathEnv ?? options?.env?.PATH },
```

Comment:
expression after `??` will never be used. I also think that when `env.PATH` is specified, we shouldn't override it